packages/components/text/src/text.styles.ts

```diff
@@ -117,6 +121,7 @@ export const detailStyles = (props: TDetailProps, theme: Theme) => css`
   ${getBaseStyles(theme)}
   display: block;
```

Comment:
Hmm I guess this is the main cause of this issue 🤔

Using `as="span"` changes the HTML semantic since we want to use `<small>` (I suppose).

I wonder why we have `display: block` in the first place though, if we're using `<small>`.

I'm thinking to actually change `small` to a `div`, then we can use `as="span"` and things are consistent. And remove `display: block`

Reply:
Hm, that would be a good proposal.
I'm leaning towards more removing just the `display` rule.
I don't think we need to change `small -> div`

Reply:
aaah dear... now I finally found a doc spec where `small` is by default an inline element